crates/autopilot/src/limit_order_quoter.rs

```diff
@@ -57,63 +58,70 @@ impl LimitOrderQuoter {
-                                sell_amount: SellAmount::BeforeFee {
-                                    value: order.data.sell_amount + order.data.fee_amount,
+            for chunk in orders.chunks(10) {
+                ignored_orders += stream::iter(chunk)
+                    .map(|order| async move {
+                        match self
+                            .quoter
+                            .calculate_quote(QuoteParameters {
+                                sell_token: order.data.sell_token,
+                                buy_token: order.data.buy_token,
+                                side: match order.data.kind {
+                                    OrderKind::Buy => OrderQuoteSide::Buy {
+                                        buy_amount_after_fee: order.data.buy_amount,
+                                    },
+                                    OrderKind::Sell => OrderQuoteSide::Sell {
+                                        sell_amount: SellAmount::BeforeFee {
+                                            value: order.data.sell_amount + order.data.fee_amount,
+                                        },
+                                    },
                                 },
-                            },
-                        },
-                        from: order.metadata.owner,
-                        app_data: order.data.app_data,
-                        signing_scheme: match order.signature {
-                            Signature::Eip712(_) => QuoteSigningScheme::Eip712,
-                            Signature::EthSign(_) => QuoteSigningScheme::EthSign,
-                            Signature::Eip1271(_) => QuoteSigningScheme::Eip1271 {
-                                onchain_order: false,
-                                verification_gas_limit: default_verification_gas_limit(),
-                            },
-                            Signature::PreSign => QuoteSigningScheme::PreSign {
-                                onchain_order: false,
-                            },
-                        },
-                    })
-                    .await
-                {
-                    Ok(quote) => {
-                        if let Err(err) = self
-                            .database
-                            .update_surplus_fee(&order.metadata.uid, quote.fee_amount)
+                                from: order.metadata.owner,
+                                app_data: order.data.app_data,
+                                signing_scheme: match order.signature {
+                                    Signature::Eip712(_) => QuoteSigningScheme::Eip712,
+                                    Signature::EthSign(_) => QuoteSigningScheme::EthSign,
+                                    Signature::Eip1271(_) => QuoteSigningScheme::Eip1271 {
+                                        onchain_order: false,
+                                        verification_gas_limit: default_verification_gas_limit(),
```

Comment:
I didn't notice this before, but we probably want to pass in the result of `SignatureValidating::validate_signature_and_get_additional_gas()` here. I.e. For ERC-1271 signatures, we need to check at each auction whether or not the signature is valid before including it. We should should change the check to check + get additional gas for limit orders and use the additional gas here.

Reply:
https://github.com/cowprotocol/services/issues/766